### PR TITLE
Don't start stopped postgres in pause

### PR DESCRIPTION
**Pull request by CyberDem0n**
Due to a race condition Patroni was falsely assuming that the standby should be restarted because some recovery parameters (primary_conninfo or similar) were changed.

Close https://github.com/zalando/patroni/issues/2834

### CODE REVIEW

File: patroni/postgresql/config.py [hunk 1]
```diff
@@ -623,7 +623,24 @@ def _recovery_parameters_to_compare(self) -> CaseInsensitiveSet:
...
-    def _read_recovery_params(self) -> Tuple[Optional[CaseInsensitiveDict], Optional[bool]]:
+    def _read_recovery_params(self) -> Tuple[Optional[CaseInsensitiveDict], bool]:
+        """Read current recovery parameters values.
+
+        .. note::
+            We query Postgres only if we detected that Postgresql was restarted
+            or when at least one of the following files was updated:
+
+                * `postgresql.conf`
+                * `postgresql.auto.conf`
```
Comment (matthbakeredb) [inline patroni/postgresql/config.py:634]:
```suggestion
                * ``postgresql.conf``;
                * ``postgresql.auto.conf``;
```

File: patroni/postgresql/config.py [hunk 2]
```diff
@@ -644,11 +661,20 @@ def _read_recovery_params(self) -> Tuple[Optional[CaseInsensitiveDict], Optional
             self._postgresql_conf_mtime = pg_conf_mtime
             self._auto_conf_mtime = auto_conf_mtime
             self._postmaster_ctime = postmaster_ctime
-        except Exception:
+        except Exception as exc:
+            if isinstance(exc, PostgresConnectionException) \
+                    and self._postgresql_conf_mtime == pg_conf_mtime \
+                    and self._auto_conf_mtime == auto_conf_mtime \
+                    and self._passfile_mtime == passfile_mtime \
+                    and self._postmaster_ctime != postmaster_ctime:
```
Comment (matthbakeredb) [inline patroni/postgresql/config.py:669]:
```suggestion
            if all(isinstance(exc, PostgresConnectionException),
                   self._postgresql_conf_mtime == pg_conf_mtime,
                   self._auto_conf_mtime == auto_conf_mtime,
                   self._passfile_mtime == passfile_mtime,
                   self._postmaster_ctime != postmaster_ctime):
```

**Review by matthbakeredb — CHANGES_REQUESTED** [review]
Seems good, couple small suggestions.

**Comment by CyberDem0n** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**